src/components/syncPlay/core/players/GenericPlayer.js

```diff
@@ -0,0 +1,305 @@
+    }
+
+    /**
+     * Called when playback starts.
+     */
+    onPlaybackStart(player, state) {
+        this.playbackCore.onPlaybackStart(player, state);
+        Events.trigger(this, 'playbackstart', [player, state]);
+    }
+
+    /**
+     * Called when playback stops.
+     */
+    onPlaybackStop(stopInfo) {
+        this.playbackCore.onPlaybackStop(stopInfo);
+        Events.trigger(this, 'playbackstop', [stopInfo]);
+    }
+
+    /**
+     * Called when playback unpauses.
+     */
+    onUnpause() {
+        this.playbackCore.onUnpause();
+        Events.trigger(this, 'unpause', [this.currentPlayer]);
+    }
+
+    /**
+     * Called when playback pauses.
+     */
+    onPause() {
+        this.playbackCore.onPause();
+        Events.trigger(this, 'pause', [this.currentPlayer]);
+    }
+
+    /**
+     * Called on playback progress.
+     * @param {Object} event The time update event.
+     * @param {Object} timeUpdateData The time update data.
+     */
+    onTimeUpdate(event, timeUpdateData) {
+        this.playbackCore.onTimeUpdate(event, timeUpdateData);
+        Events.trigger(this, 'timeupdate', [event, timeUpdateData]);
+    }
+
+    /**
+     * Called when player is ready to resume playback.
+     */
+    onReady() {
+        this.playbackCore.onReady();
+        Events.trigger(this, 'ready');
+    }
+
+    /**
+     * Called when player is buffering.
+     */
+    onBuffering() {
+        this.playbackCore.onBuffering();
+        Events.trigger(this, 'buffering');
+    }
+
+    /**
+     * Called when changes are made to the play queue.
+     */
+    onQueueUpdate() {
+        // Do nothing.
+    }
+
+    /**
+     * Gets player status.
+     * @returns {boolean} Whether the player has some media loaded.
+     */
+    isPlaybackActive() {
+        return false;
+    }
+
+    /**
+     * Gets playback status.
+     * @returns {boolean} Whether the playback is unpaused.
+     */
+    isPlaying() {
+        return false;
+    }
+
+    /**
+     * Gets playback position.
+     * @returns {number} The player position, in milliseconds.
+     */
+    currentTime() {
+        return 0;
+    }
+
+    /**
+     * Checks if player has playback rate support.
+     * @returns {boolean} _true _ if playback rate is supported, false otherwise.
+     */
+    hasPlaybackRate() {
+        return false;
+    }
+
+    /**
+     * Sets the playback rate, if supported.
+     * @param {number} value The playback rate.
+     */
+    setPlaybackRate(value) {
+        // Do nothing.
+    }
+
+    /**
+     * Gets the playback rate.
+     * @returns {number} The playback rate.
+     */
+    getPlaybackRate() {
+        return 1.0;
+    }
+
+    /**
+     * Checks if player is remotely self-managed.
+     * @returns {boolean} _true_ if the player is remotely self-managed, _false_ otherwise.
+     */
+    isRemote() {
+        return false;
+    }
+
+    /**
+     * Unpauses the player.
+     */
+    localUnpause() {
```

Comment:
Why are there so many empty methods here?

Reply:
The idea was to leave a template for reference/future players. Should I add a comment saying this?

Reply:
Ok. I think this is fine as is since we're trying to get this in for 10.7. In the future less is probably more, we can always extend things as it becomes necessary while removing unused code can be more challenging. :smile: